gossipd/gossip_store.c

```diff
@@ -77,12 +77,95 @@ static bool append_msg(int fd, const u8 *msg, u32 timestamp, u64 *len)
 	return writev(fd, iov, ARRAY_SIZE(iov)) == sizeof(hdr) + msglen;
 }
 
+/* Read gossip store entries, copy non-deleted ones.  This code is written
+ * as simply and robustly as possible! */
+static void gossip_store_compact_offline(void)
+{
+	size_t count = 0, deleted = 0;
+	int old_fd, new_fd;
+	struct gossip_hdr hdr;
+	u8 version;
+
+	old_fd = open(GOSSIP_STORE_FILENAME, O_RDONLY);
+	if (old_fd == -1)
+		return;
+	new_fd = open(GOSSIP_STORE_TEMP_FILENAME, O_RDWR|O_TRUNC|O_CREAT, 0600);
+	if (new_fd < 0) {
+		status_broken(
+		    "Could not open file for gossip_store compaction");
+		goto close_old;
+	}
+
+	if (!read_all(old_fd, &version, sizeof(version))
+	    || version != GOSSIP_STORE_VERSION) {
```

Comment:
is there an upgrade path for old versions?

Reply:
No.  This simply upgrades the current version.  And the next step deletes unknown version, but that's already-existing code

Reply:
oh nice. I should have looked a bit further perhaps. sgtm :)